ebpf_extensions/neteventebpfext/netevent_ebpf_ext_event.c

```diff
@@ -403,7 +403,7 @@ _ebpf_netevent_program_context_create(
     // Copy the event's pointer & size from the caller, to the out context.
     netevent_event_context->netevent_event_md.event_data_start = (uint8_t*)data_in;
     netevent_event_context->netevent_event_md.event_data_end = (uint8_t*)data_in + data_size_in;
-    *context = netevent_event_context;
+    *context = &netevent_event_context->netevent_event_md;
```

Comment:
For my understanding, how was it working before, with incorrect context?

Reply:
context_create was not being called in the other tests since netevent_sim directly invokes the function with payload [here ](https://github.com/microsoft/ntosebpfext/blob/671a8bad1f25694826da49402cc3ce3d6315c2d9/tests/neteventebpfext/netevent_sim/netevent_sim.c#L112)

bpf_prog_test_run does however make a call to context_create which is how we discovered the bug